source/grid/cell_id.cc

```diff
@@ -21,13 +21,110 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+
+CellId::CellId(const unsigned int dim,
+               const std::vector<char> &binary_representation)
+{
+  coarse_cell_id = 0;
+  unsigned int i = 0;
+  for (; i<sizeof(unsigned int); ++i)
+    {
+      const unsigned int temp = static_cast<unsigned int> (binary_representation[i]);
+      coarse_cell_id = coarse_cell_id | (temp<<(i*8));
+    }
+
+  unsigned int n_child_indices = 0;
+  for (unsigned int j=0; j<sizeof(unsigned int); ++i,++j)
+    {
+      const unsigned int temp = static_cast<unsigned int> (binary_representation[i]);
+      n_child_indices = n_child_indices | (temp<<(j*8));
+    }
+
+  child_indices.resize(n_child_indices);
+
+  const unsigned int children_per_char = 8 / dim;
+
+  unsigned int child_level=0;
+  const unsigned int mask = (1<<dim) - 1;
+
+  for (; i<binary_representation.size(); ++i)
+    {
+      for (unsigned int j=0; j<children_per_char; ++j)
+        {
+          child_indices[child_level] = (binary_representation[i] >> (dim*j)) & mask;
+          ++child_level;
+        }
+    }
+}
+
+template<int dim>
+std::vector<char>
+CellId::to_binary() const
+{
+  const unsigned int n_child_indices = child_indices.size();
+  const unsigned int children_per_char = 8 / dim;
+
+  const unsigned int chars_for_children = (n_child_indices / children_per_char)
+                                          + ((n_child_indices % children_per_char != 0) ? 1 : 0);
+
+  std::vector<char> binary_representation(sizeof(coarse_cell_id)+sizeof(n_child_indices)+chars_for_children,0);
+
+  unsigned int i = 0;
+  for (unsigned int j=0; j<sizeof(coarse_cell_id); ++i,++j)
+    binary_representation[i] = coarse_cell_id >> (j*8);
+  for (unsigned int j=0; j<sizeof(n_child_indices); ++i,++j)
+    binary_representation[i] = n_child_indices >> (j*8);
+
+  unsigned int child_level=0;
+  for (; i<binary_representation.size(); ++i)
+    {
+      for (unsigned int j=0; j<children_per_char; ++j)
+        {
+          const unsigned int a = static_cast<unsigned int>(child_indices[child_level]);
+          const unsigned int shifted_child_index = a << (j*dim);
+          binary_representation[i] = binary_representation[i] | shifted_child_index;
+          ++child_level;
+          if (child_level == n_child_indices)
+            break;
+        }
+    }
+
+  return binary_representation;
+}
+
+#ifdef DEAL_II_WITH_CXX11
+std::string
+CellId::to_string() const
+{
```

Comment:
Yes the result of the function is the same, but the C++11 version is significantly faster (a factor of 2 if I remember). It is a leftover of my first try to speed up the string version though, and I would not object to remove it again, now that there is the binary version.